tests/testthat/test-export.R

```diff
@@ -7,16 +7,47 @@ testthat::test_that("Listing print correctly", {
   })
 })
 
+testthat::test_that("key columns repeat with export_as_txt", {
+  # pre-processing and ordering
+  tmp_data <- ex_adae %>%
+    dplyr::slice(1:30) %>%
+    dplyr::distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)
+
+  lsting <- as_listing(tmp_data,
+                       key_cols = c("USUBJID", "AGE"),
+                       disp_cols = character()
+  ) %>%
+    add_listing_col("BMRKR1", format = "xx.x")
+
+  listing_exp <- suppressMessages(export_as_txt(lsting, lpp = 4, verbose = TRUE, page_break = "\n",
+                                                rep_cols = length(get_keycols(lsting))))
+
+  testthat::expect_snapshot(cat(listing_exp))
+})
+
+
+testthat::test_that("key columns repeat with pagination with export_as_txt", {
+  # pre-processing and ordering
+  tbl <- as_listing(
+    iris,
+    key_cols = c("Species"),
+    disp_cols = c("Sepal.Length", "Sepal.Width", "Petal.Length", "Petal.Width")
+  )
+
+  listing <- suppressMessages(export_as_txt(tbl, lpp = 20, paginate = TRUE,
+                                            page_break = "\n",
+                                            rep_cols = length(get_keycols(tbl))))
```

Comment:
I think it is better to strsplit on the original page_break and reduce the output to be snapshot